src/llvm_backend_stmt.cpp

```diff
@@ -478,10 +469,22 @@ void lb_build_range_interval(lbProcedure *p, AstBinaryExpr *node,
 	lbValue lower = lb_build_expr(p, node->left);
 	lbValue upper = {}; // initialized each time in the loop
 
-	lbAddr value = lb_add_local_generated(p, val0_type ? val0_type : lower.type, false);
+	lbAddr value;
+	if (val0_type != nullptr) {
+		Entity *e = entity_of_node(rs->vals[0]);
+		value = lb_add_local(p, val0_type, e, false);
+	} else {
+		value = lb_add_local_generated(p, lower.type, false);
+	}
 	lb_addr_store(p, value, lower);
 
-	lbAddr index = lb_add_local_generated(p, t_int, false);
+	lbAddr index;
+	if (val1_type != nullptr) {
+		Entity *e = entity_of_node(rs->vals[1]);
+		index = lb_add_local(p, t_int, e, false);
```

Comment:
This should be `e->type` not `t_int`

Reply:
all set

Reply:
Still wrong because you still have `lb_add_local_generated(p, t_int, false)`

Reply:
I'm confused... Line 486 from the diff? Wouldn't it need a fallback type when there's no entity for the iteration value?

Reply:
Sorry, I read it wrong 😆

Reply:
all good 👍